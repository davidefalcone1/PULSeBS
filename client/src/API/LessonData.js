import moment from 'moment';

class LessonData {
    constructor(scheduleId, courseId, startDate, endDate, occupiedSeats, availableSeats,
        isLessonCancelled, isLessonRemote, classroom
        /*, normalBookings, cancelledBookings, waitingBookings */) {
        if (scheduleId)
            this.scheduleId = scheduleId;
        this.courseId = courseId;
        this.startDate = moment(new Date(startDate));
        this.endDate = moment(new Date(endDate));
        this.occupiedSeats = occupiedSeats;
        this.availableSeats = availableSeats;
        this.isLessonCancelled = isLessonCancelled;
        this.isLessonRemote = isLessonRemote;
        this.classroom = classroom

        //TODO use the correct variables
        this.normalBookings = 0;
        this.cancelledBookings = 0;
        this.waitingBookings = 0;
    }

    static fromJson(json) {
        const temp = Object.assign(new LessonData(), json);
        temp.startDate = moment(new Date(temp.startDate));
        temp.endDate = moment(new Date(temp.endDate));
        return temp;
    }
}

export default LessonData;