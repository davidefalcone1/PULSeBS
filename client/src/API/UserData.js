class UserData{
    constructor(id, personId, fullName, email) {
        if (id) 
            this.id = id;
        this.personId = personId;
        this.fullName = fullName;
        this.email = email;
    }

    static fromJson(json){
        const temp =  Object.assign(new UserData(), json);
        return temp;
    }
}

export default UserData;