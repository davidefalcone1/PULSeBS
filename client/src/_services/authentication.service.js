import { BehaviorSubject } from 'rxjs';


const currentUserSubject = new BehaviorSubject(JSON.parse(localStorage.getItem('user')));

export const authenticationService = {
    login,
    logout,
    user: currentUserSubject.asObservable(),
    get currentUserValue () { return currentUserSubject.value }
};

function handleResponse(response) {
    return response.text().then(text => {
        const data = text && JSON.parse(text);
        if (!response.ok) {
            if ([401, 403].indexOf(response.status) !== -1) {
                // auto logout if 401 Unauthorized or 403 Forbidden response returned from api
                authenticationService.logout();
            }

            const error = (data && data.message) || response.statusText;
            return Promise.reject(error);
        }

        return data;
    });
}

function login(username, password) {
    const requestOptions = {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ username, password })
    };
    return new Promise(async function (resolve, reject) {
        try {
            const response = await fetch('/users/authenticate', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify({ username, password }),
            });
            const user = await response.json().then(handleResponse);
            if (response.ok) {
                localStorage.setItem('user', JSON.stringify(user));
                currentUserSubject.next(user);
                resolve(user);
            }
            else
                reject();
        }
        catch (e) {
            reject();
        }
    });

}
function logout() {

    const requestOptions = {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
    };
    localStorage.removeItem('user');
    currentUserSubject.next(null);
    return fetch('/logout',requestOptions);
}