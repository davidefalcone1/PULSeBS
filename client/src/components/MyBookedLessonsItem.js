import React from 'react';
import ListGroup from 'react-bootstrap/ListGroup';
import Button from 'react-bootstrap/Button';

const lessonListItem = (props) => {
  return (
    <ListGroup.Item id = {"lesson-" + props.lesson.id}>
        <div className="d-flex w-100 pt-3 justify-content-between no-gutters">
                <StartingTimeField id = {props.lesson.scheduleId} startingTime = {props.lesson.startingTime}/>
                <EndingTimeField id = {props.lesson.scheduleId} endingTime = {props.lesson.endingTime}/>
                <BookingStatusField id = {props.lesson.scheduleId} occupiedSeats = {props.lesson.occupiedSeats} availableSeats = {props.lesson.availableSeats}/>
                <SelectField id = {props.lesson.scheduleId} selectLessonFunction = {props.selectLessonFunction} />
        </div>
    </ListGroup.Item>
  );
}

function StartingTimeField(props){
    return(
        <div className="col-sm-3">
            <p id={"startingTimeOfLesson" + props.id}>
                {props.startingTime.format("ddd DD-MM-YYYY HH:mm").toString()}
            </p>
        </div>
    );
}

function EndingTimeField(props){
    return(
        <div className="col-sm-3">
            <p id={"endingTimeOfLesson" + props.id}>
                {props.endingTime.format("ddd DD-MM-YYYY HH:mm").toString()}
            </p>
        </div>
    );
}

function BookingStatusField(props){
    return(
        <div className="col-sm-3">
            <p id={"seatsStatusOfLesson" + props.id}>
                {props.occupiedSeats} / {props.availableSeats}
            </p>
        </div>
    );
}

function SelectField(props){
    return(
        <div className="col-sm-3">
            <Button variant="danger" onClick={(event) => {
                event.preventDefault();
                props.selectLessonFunction(props.id);               
            }} id={"deleteFieldOfLesson" + props.id}>
                DELETE
            </Button>
        </div>
    );
}

export default lessonListItem;