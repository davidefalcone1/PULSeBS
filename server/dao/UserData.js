class UserData{
    constructor(id, personId, fullName, email) {
        if (id) 
            this.id = id;
        this.personId = personId;
        this.fullName = fullName;
        this.email = email;
    }

}

module.exports = UserData;