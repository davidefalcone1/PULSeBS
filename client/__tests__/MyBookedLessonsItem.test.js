import React from 'react';
import renderer from 'react-test-renderer';
import { AuthContext } from '../src/_services/AuthContext';
import MyBookedLessonsItem from '../src/components/MyBookedLessonsItem';
import { BrowserRouter } from 'react-router-dom';
import UserData from '../src/API/UserData';
import LessonData from '../src/API/LessonData';
import moment from 'moment';

Date.now = jest.fn(() => 1482363367071);//Avoid dynamic content on the page

describe('MyBookedLessonsItem', ()=>{
    test('Renders correctly', ()=>{
        const component = renderer.create(
            <BrowserRouter>
                <AuthContext.Provider value={{user: new UserData(1, '12345', 'Davide Falcone', 'davide.falcone@polito.it')}}>
                    <MyBookedLessonsItem lesson={new LessonData(1, 'XY123', moment().add(1, 'days'), moment().add(1, 'days').add(1, 'hours'), 9, 90, false, true, 'A1', 100, 4, 33, 0)} selectLessonFunction={()=>{}}/>
                </AuthContext.Provider>
            </BrowserRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});