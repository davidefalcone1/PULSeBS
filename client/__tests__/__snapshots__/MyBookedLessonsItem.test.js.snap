// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyBookedLessonsItem Renders correctly 1`] = `
<div
  className="list-group-item"
  disabled={false}
  id="lesson-undefined"
  onClick={[Function]}
>
  <div
    className="d-flex w-100 pt-3 justify-content-between no-gutters"
  >
    <div
      className="col-sm-3"
    >
      <p
        id="startDateOfLesson1"
      >
        Fri 23-12-2016 00:36
      </p>
    </div>
    <div
      className="col-sm-3"
    >
      <p
        id="endDateOfLesson1"
      >
        Fri 23-12-2016 01:36
      </p>
    </div>
    <div
      className="col-sm-3"
    >
      <p
        id="seatsStatusOfLesson1"
      >
        9
         / 
        90
      </p>
    </div>
    <div
      className="col-sm-3"
    >
      <button
        className="btn btn-danger"
        disabled={false}
        id="deleteFieldOfLesson1"
        onClick={[Function]}
        type="button"
      >
        DELETE
      </button>
    </div>
  </div>
</div>
`;
